presto-main/src/main/java/com/facebook/presto/operator/aggregation/histogram/SingleTypedHistogram.java

```diff
@@ -0,0 +1,343 @@
+
+        checkArgument(expectedSize > 0, "expectedSize must be greater than zero");
+
+        maxFill = calculateMaxFill(hashCapacity);
+        mask = hashCapacity - 1;
+        hashPositions = new IntBigArray(-1);
+        hashPositions.ensureCapacity(hashCapacity);
+        counts = new LongBigArray();
+        counts.ensureCapacity(hashCapacity);
+    }
+
+    public SingleTypedHistogram(Type type, int expectedSize)
+    {
+        this(type, expectedSize, computeBucketCount(expectedSize), type.createBlockBuilder(null, computeBucketCount(expectedSize)));
+    }
+
+    private static int computeBucketCount(int expectedSize)
+    {
+        return arraySize(expectedSize, FILL_RATIO);
+    }
+
+    public SingleTypedHistogram(Block block, Type type, int expectedSize)
+    {
+        this(type, expectedSize);
+        requireNonNull(block, "block is null");
+        for (int i = 0; i < block.getPositionCount(); i += 2) {
+            add(i, block, BIGINT.getLong(block, i + 1));
+        }
+    }
+
+    @Override
+    public long getEstimatedSize()
+    {
+        return INSTANCE_SIZE + values.getRetainedSizeInBytes() + counts.sizeOf() + hashPositions.sizeOf();
+    }
+
+    @Override
+    public void serialize(BlockBuilder out)
+    {
+        if (values.getPositionCount() == 0) {
+            out.appendNull();
+        }
+        else {
+            Block valuesBlock = values.build();
+            BlockBuilder blockBuilder = out.beginBlockEntry();
+            for (int i = 0; i < valuesBlock.getPositionCount(); i++) {
+                type.appendTo(valuesBlock, i, blockBuilder);
+                BIGINT.writeLong(blockBuilder, counts.get(i));
+            }
+            out.closeEntry();
+        }
+    }
+
+    @Override
+    public void addAll(TypedHistogram other)
+    {
+        other.readAllValues(
+                (block, position, count) -> add(position, block, count));
+    }
+
+    @Override
+    public void readAllValues(HistogramValueReader reader)
+    {
+        for (int i = 0; i < values.getPositionCount(); i++) {
+            long count = counts.get(i);
+            if (count > 0) {
+                reader.read(values, i, count);
+            }
+        }
+    }
+
+    @Override
+    public void add(int position, Block block, long count)
+    {
+        int hashPosition = getBucketId(TypeUtils.hashPosition(type, block, position), mask);
+        hashStats.incInitHash();
+        Supplier<Void> fuu = Suppliers.memoize(hashStats::incRehashCollisionOnly);
+
+        // look for an empty slot or a slot containing this key
+        while (true) {
+            if (hashPositions.get(hashPosition) == -1) {
+                break;
+            }
+
+            if (type.equalTo(block, position, values, hashPositions.get(hashPosition))) {
+                counts.add(hashPositions.get(hashPosition), count);
+                return;
+            }
+            hashStats.incInitCollision();
+            fuu.get();
```

Comment:
`fuu`?

Reply:
yea, found these and got rid of all hash stats